enterprise/models.py

```diff
@@ -1686,6 +1691,74 @@ def sync_learner_profile_data(self):
         return identity_provider is not None and identity_provider.sync_learner_profile_data
 
 
+class EnterpriseCourseEntitlementManager(models.Manager):
+    """
+    Model manager for `EnterpriseCourseEntitlement`.
+    """
+
+    def get_queryset(self):
+        """
+        Override to return only those entitlment records for which learner is linked to an enterprise.
+        """
+        return super().get_queryset().select_related('enterprise_customer_user').filter(
+            enterprise_customer_user__linked=True
+        )
+
+
+class EnterpriseCourseEntitlement(TimeStampedModel):
+    """
+    Store the information about the entitlement of an enterprise user for a course
+    """
+
+    objects = EnterpriseCourseEntitlementManager()
+
+    class Meta:
+        unique_together = (('enterprise_customer_user', 'course_uuid',),)
+        app_label = 'enterprise'
+        ordering = ['created']
+
+    uuid = models.UUIDField(
+        primary_key=True,
+        editable=False,
+        null=False,
+        default=uuid4,
+    )
+    course_uuid = models.CharField(
```

Comment:
We probably want this to be a content key and not a UUID.

Reply:
b2c entitlement's specifically reference course uuid instead of key, I could change this to content key but then there would have to be another level of work done so that we can connect the two records.

Reply:
ah, great point.  Then yeah, keep `course_uuid`.